awx/main/management/commands/inventory_import.py

```diff
@@ -268,13 +268,18 @@ def _build_db_instance_id_map(self):
         """
         self.db_instance_id_map = {}
         if self.instance_id_var:
-            host_qs = self.inventory_source.hosts.all()
-            host_qs = host_qs.filter(instance_id='', variables__contains=self.instance_id_var.split('.')[0])
+            host_qs = self.inventory_source.hosts.all().only('id', 'instance_id', 'variables').iterator()
```

Comment:
I think we need an alternative to this for performance reasons.

To do that, I don't see easy option other than to do a loop of `instance_id_var.split(',')` (probably reversed), and then `.split('.')[0]` inside that loop.